tests/test_framework.py

```diff
@@ -452,6 +459,84 @@ def test_resource_dsl_parsing(self):
+                                                    resource_plugin='test',
+                                                    resource_creation_timestamp=_TIMESTAMP,
+                                                    resource_ttl=RESOURCE_MANAGER_RESOURCE_EXPIRE)
+        self.my_dir, self.panoptes_test_conf_file = _get_test_conf_file()
+
+    @patch('redis.StrictRedis', panoptes_mock_redis_strict_client)
+    def test_resource_cache(self):
+        panoptes_context = PanoptesContext(self.panoptes_test_conf_file,
+                                           key_value_store_class_list=[PanoptesTestKeyValueStore])
+
+        #  Test PanoptesResourceCache methods when setup_resource_cache not yet called
+        panoptes_resource_cache = PanoptesResourceCache(panoptes_context)
+        test_query = 'resource_class = "network"'
+        with self.assertRaises(PanoptesResourceError):
+            panoptes_resource_cache.get_resources(test_query)
+        panoptes_resource_cache.close_resource_cache()
+
+        panoptes_resource = self.__panoptes_resource
+        panoptes_resource.add_metadata("metadata_key1", "test")
+        panoptes_resource.add_metadata("metadata_key2", "test")
+
+        kv = panoptes_context.get_kv_store(PanoptesTestKeyValueStore)
+        serialized_key, serialized_value = PanoptesResourceStore._serialize_resource(panoptes_resource)
+        kv.set(serialized_key, serialized_value)
+
+        mock_kv_store = Mock(return_value=kv)
+
+        with patch('yahoo_panoptes.framework.resources.PanoptesContext.get_kv_store', mock_kv_store):
+            # Test errors when setting up resource cache
+            mock_resource_store = Mock(side_effect=Exception)
+            with patch('yahoo_panoptes.framework.resources.PanoptesResourceStore', mock_resource_store):
+                with self.assertRaises(PanoptesResourceError):
+                    panoptes_resource_cache.setup_resource_cache()
+
+            mock_connect = Mock(side_effect=Exception)
+            with patch('yahoo_panoptes.framework.resources.sqlite3.connect', mock_connect):
+                with self.assertRaises(PanoptesResourceError):
+                    panoptes_resource_cache.setup_resource_cache()
+
+            mock_get_resources = Mock(side_effect=Exception)
+            with patch('yahoo_panoptes.framework.resources.PanoptesResourceStore.get_resources', mock_get_resources):
+                with self.assertRaises(PanoptesResourceError):
+                    panoptes_resource_cache.setup_resource_cache()
+
+            # Test basic operations
+            panoptes_resource_cache.setup_resource_cache()
+            self.assertIsInstance(panoptes_resource_cache.get_resources('resource_class = "network"'),
+                                  PanoptesResourceSet)
+            self.assertEqual(0, len(panoptes_resource_cache.get_resources('resource_class = "network"')))
```

Comment:
The pattern we've been using in other tests is assert(returnedValue, expectedValue) - so this should look like: 
self.assertEqual(len(panoptes_resource_cache.get_resources('resource_class = "network"')), 0)